prover/protocol/distributed/distributed.go

```diff
@@ -69,10 +76,56 @@ func extractDistModule(
 	moduleName moduleName,
 	maxSegmentSize, maxNumSegment int,
 ) DistributedModule {
+	// initialize  two compiledIOPs, for LPP and GL.
+	disModule := DistributedModule{
+		LookupPermProj: &wizard.CompiledIOP{},
+		GlobalLocal:    &wizard.CompiledIOP{},
+	}
+
+	for _, qName := range comp.QueriesNoParams.AllUnignoredKeys() {
+
+		// Filter LPP queries
+		q := comp.QueriesNoParams.Data(qName)
+
+		switch v := q.(type) {
```

Comment:
Since you have a switch, it would be simpler to have one function for global, one for local and one for lookup, and one for projection and one for permutation. That way you don't have to do a switch in a switch, it's nicer to read that way.

You may want to move the `if disc.QueryIsInModule(v, moduleName)` our of the switch as it happens in all situations